### PR TITLE
Fix pylinting single file

### PR DESCRIPTION
**Pull request by leohemsted**
as pointed out [here](https://github.com/AtomLinter/linter-pylint/issues/80#issuecomment-191534104), opening a single file leads to problems as projDir is set to `[...]/dir/my_file.py` instead of `[...]/dir`.

i'm barely familiar with atom packages or coffeescript so let me know if there's any better way to solved this

also it works on windows (64bit) - will test on ubuntu in a bit, don't have an OSX install to hand to test on that.

---
This change is [Reviewable](https://reviewable.io/reviews/atomlinter/linter-pylint/136)

### CODE REVIEW

**Comment by Arcanemagus** [comment]
Travis is testing on OS X btw :wink:

File: lib/main.coffee [hunk 1]
```diff
@@ -105,7 +105,8 @@ module.exports =
                 lintIssue
 
   getProjDir: (filePath) ->
-    atom.project.relativizePath(filePath)[0]
+    projDir = atom.project.relativizePath(filePath)[0]
+    if projDir is filePath then path.parse(filePath).dir else projDir
```
Comment (Arcanemagus) [inline lib/main.coffee]: Might be clearer to replace the `path.parse(filePath).dir` with `path.dirname(filePath)`?

**Comment by leohemsted** [comment]
@Arcanemagus updated use of `dirname` - i realise travis' testing on OS X but a brief look at the test appears that we're not, at least, trying out loading a project folder vs file - so i'd like it if someone could either gimme a hand/some pointers writing a relevant test (I've never done anything with the atom package api), or just fire it up on OSX and make sure i haven't somehow broken everything. I'm not ruling anything out when it comes to platform independency :sob:

**Comment by Arcanemagus** [comment]
I'm going to mark this as closed for a few reasons:

- The codebase is now ES6, so this would need to be redone in the first place
- The ability to open a file in that broken matter is an Atom bug, I don't think putting a workaround in here for that is the best of ideas
- I can't reproduce the bug anyway 😛

If you can reproduce it on Atom v1.9.0-beta0 can you contact me on Atom's Slack to debug it further?

**Comment by leohemsted** [comment]
Just checked on 1.9.0-beta0, can't reproduce as can't open the file in the same previous way (it now automatically opens the parent dir when you open file), so happy with this being closed :+1:

**Pull request closed without merging**